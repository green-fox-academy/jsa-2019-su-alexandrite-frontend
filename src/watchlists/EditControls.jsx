import React from 'react';
import { FontAwesome5 } from '@expo/vector-icons';
import {
  View,
  Text,
  TouchableHighlight,
} from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const EditorControls = ({
  checkedItems,
  isInEditMode,
  toggleEditMode,
  onDeleteWatchlist,
}) => {
  const num = checkedItems.filter((checked) => checked).length;

  return (
    <View style={styles.editorMode}>
      <TouchableHighlight style={styles.editorButton} underlayColor="#ffebeb">
        <Text style={styles.editDeleteButton}>
          <FontAwesome5 size={14} name="trash" />
          {` ${num}`}
        </Text>
      </TouchableHighlight>
      <View style={styles.editorModeRight}>
        <TouchableHighlight
          style={styles.editorButton}
          underlayColor="#ffebeb"
          onPress={onDeleteWatchlist}
        >
          <Text style={{ color: '#c12424', fontSize: 12, fontWeight: 'bold' }}> Delete WatchList</Text>
        </TouchableHighlight>
        <TouchableHighlight style={styles.editorButton} onPress={() => toggleEditMode(!isInEditMode)} underlayColor="#eee">
          <Text style={{ color: '#666', fontSize: 12, fontWeight: 'bold' }}>Cancel</Text>
        </TouchableHighlight>
      </View>
    </View>
  );
};

EditorControls.propTypes = {
  isInEditMode: PropTypes.bool.isRequired,
  toggleEditMode: PropTypes.func.isRequired,
  checkedItems: PropTypes.arrayOf(
    PropTypes.bool,
  ).isRequired,
  onDeleteWatchlist: PropTypes.func.isRequired,
};

export default EditorControls;
