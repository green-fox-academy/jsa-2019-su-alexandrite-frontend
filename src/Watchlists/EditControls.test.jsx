import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import EditControls from './EditControls';

jest.useFakeTimers();

describe('<EditControls />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = renderer.create(
      <EditControls
        checkedItems={[]}
        isInEditMode={false}
        toggleEditMode={jest.fn()}
        onDeleteWatchlist={jest.fn()}
        onDeleteStocks={jest.fn()}
      />,
    );
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const tree = renderer.create(
      <EditControls
        checkedItems={[]}
        isInEditMode={false}
        toggleEditMode={jest.fn()}
        onDeleteWatchlist={jest.fn()}
        onDeleteStocks={jest.fn()}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should toggle edit mode when tapping on editorButton', () => {
    const toggleEditMode = jest.fn();
    const toggle = jest.fn(() => toggleEditMode);
    const wrapper = shallow(
      <EditControls
        checkedItems={[]}
        isInEditMode={false}
        toggleEditMode={toggle}
        onDeleteWatchlist={() => { }}
        onDeleteStocks={() => { }}
      />,
    );
    wrapper
      .find({ testID: 'editor-button' })
      .first()
      .simulate('press');
    expect(toggle).toHaveBeenCalledWith(true);
  });
});
