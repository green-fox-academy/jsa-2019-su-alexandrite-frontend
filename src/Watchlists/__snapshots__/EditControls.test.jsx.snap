// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditControls /> Should render and match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 6,
        "paddingHorizontal": 10,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#b40d0d",
          "fontSize": 12,
          "fontWeight": "bold",
          "textAlignVertical": "center",
        }
      }
    >
      <Text />
       0
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 6,
          "paddingHorizontal": 10,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#c12424",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
         Delete WatchList
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 6,
          "paddingHorizontal": 10,
          "paddingVertical": 8,
        }
      }
      testID="editor-button"
    >
      <Text
        style={
          Object {
            "color": "#666",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</View>
`;
